### DISCUSSION TITLE
Changing/deleting lines on graph

### DISCUSSION THREAD

**Discussion by szabzola** [hnolCol/instantclue#55] [Ideas]
Hi,
I do not find any way to modify or remove line (horizontal, vertical etc. from Model) on graph.
If I make a mistake, the only way I found, is to redraw the graph, losing all other settings (other lines, colors, axis etc.) which is not convenient. I can format the line on Axis/Curves settings, but e.g. a remove button there would be beneficial.

By the way, some way to save a graph and its settings (e.g. as it is uploaded to server), which can be reloaded and modified would make our life less stressed:-). If I find out (or reviewer requests) that something has to be changed, it would be highly appreciated, if reproduction and modification of it would not need to repeat all steps and settings (especially, if I don not remember all of them :-).
Simple cosmetics, of course I can make on vector export.
Thanks anyway for the nice work again,
Zoltan

**Comment by hnolCol** [comment]
Dear Zoltan, 

thanks for the input and it is obviously a very good point :). I will add a table with added lines similar to the Color and Size tables allowing you to hide/show. Probably the possibility to edit the name and add a legend would be beneficial?

For the second part, what do you think about having the option to save a session (improved version compared to now!) with and without storing the current settings in it.Maybe it would be useful to ask the user if he/she wants to import the settings from the stored session, if they are available, or keep going with the current once. Let me know what you think - I will add these as milestones. :)

**Reply by szabzola** [reply]
1. Thanks. Changing the name of lines is possible in settings, isn't it? Optional legend may be useful also.
2. Yes, that may be useful. Saving a session with optional current graph setting is OK. I know, there are a lot of things to save, to reproduce a graph (including quick select, color, size categories, axis settings etc.), I just thought, if uploading to server is possible in some format, local saving and reloading of a specific graph format, or template could be "easily" implemented, using the same data frames as original. A more general solution could be, specifying what to use from actual dataframes if "template" is loaded with different data, instead of original numeric and categorical data etc. But just saving graph-type-specific, and matplotlib, axis and additional analysis/model line settings as template, and adding data, can also be OK.
Just ideas, it is already a wonderful work,
thanks
